### PR TITLE
feat(list-users): Implement dispatch count for list users

### PR DESCRIPTION
**Pull request by poovamraj**
## Description
Implemented DispatchCount metrics for list users. This will be useful when throttling is implemented. Since we don't have the throttling PR merged, that hasn't been implemented but it should be simple since we have the counter logic now.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by poovamraj** [comment]
@willvedd some tests are failing. Can we pair over this tomorrow?

**Comment by poovamraj** [comment]
Fixed the failing tests. It was due to uninitialized value in reflexive userset edge case.

File: tests/listusers/listusers_test.go [hunk 1]
```diff
@@ -259,7 +259,8 @@ type document
 			require.NotEmpty(t, fields["trace_id"])
 			if !test.expectedError {
 				require.NotEmpty(t, fields["datastore_query_count"])
-				require.Len(t, fields, 13)
+				require.Empty(t, fields["dispatch_count"])
```
Comment (willvedd) [inline tests/listusers/listusers_test.go:262]: Shouldn't this be _not_ empty?
```suggestion
				require.NotEmpty(t, fields["dispatch_count"])
```

**Review by willvedd — CHANGES_REQUESTED** [review]

File: tests/listusers/listusers_test.go (hunk 1 above)
Comment (willvedd) [reply tests/listusers/listusers_test.go]: This was incorrectly considered empty because it was zero. I fixed this test to incur a DB query and dispatch call. These assertions are now a bit stronger.
